### ISSUE TITLE
Rabbitmq inital connection does not re-try

### ISSUE THREAD

**Issue by reptillicus** [minio/minio#6807]
## Expected Behavior

Notifications services should retry connections upon startup of minio server. 
## Current Behavior

When starting minio server, if rabbitmq is not running and listening yet the notification system errors out and will not attempt to reconnect. This leads to strange errors with the events api with `mc`. This happens for instance when running minio and rabbitmq with docker-compose. 
```API: SYSTEM()
minio_1       | Time: 19:35:25 UTC 11/12/2018
minio_1       | Error: dial tcp 172.18.0.6:5672: connect: connection refused
minio_1       |        1: cmd/config-current.go:626:cmd.getNotificationTargets()
minio_1       |        2: cmd/notification.go:462:cmd.NewNotificationSys()
minio_1       |        3: cmd/server-main.go:380:cmd.serverMain()
minio_1       | Endpoint:  http://172.18.0.7:9000  http://127.0.0.1:9000
```
## Possible Solution

Have the notification services retry connections upon startup. 

## Steps to Reproduce (for bugs)

```version: '3'
services:
  minio:
    image: minio/minio
    ports:
      - 9000:9000
    volumes:
      - "/data:/data"
    environment:
      MINIO_ACCESS_KEY: "admin"
      MINIO_SECRET_KEY: "password"
    command: server /data
  rabbitmq:
      image: rabbitmq:3.6.5-management
      ports:
        - 5672:5672

```
`docker-compose up` on that will likely reproduce the bug. 
## Context

Basically impossible to run minio and rabbitmq with docker-compose. 
## Your Environment

* Version used (`minio version`):
Version: 2018-11-06T01:01:02Z
Release-Tag: RELEASE.2018-11-06T01-01-02Z
Commit-ID: eb1f9c9916cdd2d1031b76dfc4e539fd643de6bc
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
Linux 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

* Link to your project:

**Comment by kannappanr** [comment]
@reptillicus Thanks for filing this issue. We will try to address this issue soon.

**Comment by nitisht** [comment]
@reptillicus we have added a fix, do you mind testing with `minio/minio:edge` image and confirm if it works for you now

**Comment by reptillicus** [comment]
Absolutely, thanks!
